src/frontend/js/components/CourseAddToWishlist/index.tsx

```diff
@@ -0,0 +1,128 @@
+  },
+  [CourseAddToWishlistMessageIds.LOG_ME]: {
+    id: 'components.CourseAddToWishlist.logMe',
+    defaultMessage: 'Log in to be notify',
+    description: 'label to proceed to login page before beeing adding a course to user wishlist.',
+  },
+});
+
+enum ComponentStates {
+  INITIALIZING = 'initializing',
+  LOADING = 'loading',
+  IDLE = 'idle',
+  ERROR = 'error',
+}
+
+export interface Props {
+  courseCode: Joanie.Course['code'];
+}
+
+const CourseAddToWishlist = ({ courseCode }: Props) => {
+  const [componentState, setComponentState] = useState<ComponentStates>(
+    ComponentStates.INITIALIZING,
+  );
+  const [isInWishlist, setIsInWishlist] = useState<boolean>(false);
+  const [errorMessageId, setErrorMessageId] = useState<CourseAddToWishlistMessageIds>();
+
+  const { user, login } = useSession();
+  const wishlistCourse = useUserWishlistCourses({ courseCode });
+
+  useEffect(() => {
+    if (componentState === ComponentStates.INITIALIZING) {
+      if (!wishlistCourse.states.fetching) {
+        setIsInWishlist(wishlistCourse.items.length > 0);
+        setComponentState(ComponentStates.IDLE);
+      }
+    }
+  }, [wishlistCourse.states.fetching, componentState]);
+
+  const removeFromWishlist = () =>
+    wishlistCourse.methods.delete(courseCode, {
+      onSuccess: () => setIsInWishlist(false),
+      onError: () => {
+        setComponentState(ComponentStates.ERROR);
+        setErrorMessageId(CourseAddToWishlistMessageIds.ERROR_ADD);
+      },
+    });
+  const addToWishlist = () =>
+    wishlistCourse.methods.create(courseCode, {
+      onSuccess: () => setIsInWishlist(true),
+      onError: () => {
+        setComponentState(ComponentStates.ERROR);
+        setErrorMessageId(CourseAddToWishlistMessageIds.ERROR_REMOVE);
+      },
+    });
+
+  if (componentState === ComponentStates.INITIALIZING) return <Spinner />;
+  return user === undefined ? (
+    <Button
+      color="primary"
+      onClick={login}
+      className="user-wishlist-button"
+      disabled={componentState === ComponentStates.LOADING}
+      {...(componentState === ComponentStates.ERROR && {
+        'aria-describedby': 'user-wishlist-error',
+      })}
+    >
+      <FormattedMessage {...messages[CourseAddToWishlistMessageIds.LOG_ME]} />
+    </Button>
+  ) : (
+    <>
+      <Button
+        color="primary"
+        onClick={isInWishlist ? removeFromWishlist : addToWishlist}
+        className="user-wishlist-button"
+        disabled={componentState === ComponentStates.LOADING}
+        {...(componentState === ComponentStates.ERROR && {
+          'aria-describedby': 'user-wishlist-error',
+        })}
+      >
+        <FormattedMessage {...messages[isInWishlist ? 'labelRemove' : 'labelAdd']} />
+      </Button>
+      <p className="user-wishlist-button__error" id="user-wishlist-error" tabIndex={-1}>
+        {componentState === ComponentStates.ERROR && errorMessageId && (
+          <FormattedMessage {...messages[errorMessageId]} />
+        )}
```

Comment:
What do you think to display a generic error message in the case the component is in Error but not errorMessage has been set.

Reply:
I used api error messages.Is that ok ?